rustls/examples/internal/bogo_shim.rs

```diff
@@ -805,17 +804,6 @@ fn exec(opts: &Options, mut sess: Connection, count: usize) {
             }
         }
 
-        if !sess.is_handshaking()
-            && !opts
-                .expect_quic_transport_params
-                .is_empty()
-        {
-            let their_transport_params = sess
-                .quic_transport_parameters()
-                .expect("missing peer quic transport params");
-            assert_eq!(opts.expect_quic_transport_params, their_transport_params);
-        }
```

Comment:
do we understand when/why this became unused? because, at some point in the past, it was added for a reason to give QUIC test coverage against an independent implementation. seems like that is valuable to have?

Reply:
@djc Can you share more about how you reached the conclusion the shim code for quic should be removed?

The upstream BoringSSL [shim for bogo](https://github.com/google/boringssl/blob/92de195169d26d9f5cec7ef34df9194e614e50f8/ssl/test/bssl_shim.cc#L522-L533), and [the runner](https://github.com/google/boringssl/blob/92de195169d26d9f5cec7ef34df9194e614e50f8/ssl/test/runner/runner.go#L1456-L1498) both seem to have quic specific test coverage.

In our `config.json` I do see that Quic related bits are in the `DisabledTests`:
https://github.com/rustls/rustls/blob/fb9e181a1a777bc2f42bd7c172be1ebddb3b68fa/bogo/config.json#L110-L111

https://github.com/rustls/rustls/blob/fb9e181a1a777bc2f42bd7c172be1ebddb3b68fa/bogo/config.json#L118-L120

Is it fair to summarize that the upstream does support Quic coverage, but that we've been ignoring it and so now it makes more sense to remove the unused shim support vs adapting it to the new API to remain unused, or investing in getting it running?

Reply:
Went crazy with git blame for a bit:

- 2021-02-06 eb191d0cc4dbc36b6bc679d66cd12788461545f8 (#475, BoringSSL update): no changes
- 2020-04-12 339923fc5314d154b0522eb2ba07fd20ca98626d (Get new bogo version working): disabled more QUIC tests
- 2019-01-13 181765111cff8fe0c99e7f53670cfd971b3eb14d (Disable QUIC bogo tests): disabled the tests
- 2019-01-13 d766c4d8b6c03a2651aa13a01367232d8e2b14a7 (#187, Raw handshake I/O for QUIC): initially enabled QUIC tests in the bogo shim
- 2018-05-14 87ab639360d72ef4b45da1f3898f6d60d4316d3f (#168, enable QUIC tests in bogo): add some QUIC support in the bogo shim

So, while there initially was some QUIC support added in #187, that doesn't seem to have lasted very long, and it has been disabled ever since. This PR then merely rips out some of the scaffolding that was in place.

Reply:
hm we need to look into the commit privileges of these guys! 😨